### DISCUSSION TITLE
What is the meaning of $

### DISCUSSION THREAD

**Discussion by realyukii** [iBotPeaches/Apktool#3307] [Q&A] [answered]
I am new to Smali, and seeing so many $ in file naming:
![image](https://github.com/iBotPeaches/Apktool/assets/126035476/ab2c1e23-ca8e-46a8-b8a8-9c164614fc5d)

**Comment by iBotPeaches** [comment, marked as answer]
They normally denote a child/inner class. I can't see the context/file to be sure, but thats generally the reason.

Its easier to see this with `R.java`

```java
public final class R {  
    public static final class attr {  
    }  
    public static final class drawable {  
        public static final int ic_launcher=0x7f020000;  
    }
    // ...
}
```

```text
➜ ls
'R$color.smali'  'R$drawable.smali'  'R$mipmap.smali'  'R$string.smali'  'R$xml.smali'   R.smali
```

**Reply by realyukii** [reply]
Thanks for the answer! :)
